share/getters/cascade.go

```diff
@@ -119,13 +120,21 @@ func cascadeGetters[V any](
 		}
 	}()
 
+	minTimeout := time.Duration(0)
+	_, ok := ctx.Deadline()
+	if !ok {
+		// in this case minTimeout will be applied for all getters,so each of them
+		// will have 1 minute timeout.
+		minTimeout = time.Minute
+	}
```

Comment:
Well, we still may encounter case, when user set 1Hour timeout here.
Perhaphs limiting timeout to be strict max 3min would be simple and robust. 
```
ctx = WithTimeout(ctx,3min)
```

Reply:
> Well, we still may encounter case, when user set 1Hour timeout here. 

True, but then the user explicitly wants to wait 1h. I wanted to cover the case where he hasn't specified anything at all.